addons/website/static/src/client_actions/website_editor/website_editor.js

```diff
@@ -44,12 +45,13 @@ export class WebsiteEditorClientAction extends Component {
                 this.websiteService.contentWindow = this.iframe.el.contentWindow;
 
                 this.iframe.el.contentWindow.addEventListener('beforeunload', () => {
+                    this.websiteService.blockIframe();
                     if (!this.websiteContext.edition) {
                         this.iframefallback.el.contentDocument.body.replaceWith(this.iframe.el.contentDocument.body.cloneNode(true));
                         $().getScrollingElement(this.iframefallback.el.contentDocument)[0].scrollTop = $().getScrollingElement(this.iframe.el.contentDocument)[0].scrollTop;
                     }
                 });
-
+                this.websiteService.unblockIframe();
```

Comment:
to confine the blockUI for the first loading only, we could use another useEffect that only takes care of that:
```js
        useEffect(() => {
            this.websiteService.blockIframe();
            this.iframe.el.addEventListener('load', () => this.websiteService.unblockIframe(), { once: true });
        }, () => []);
```